### PR TITLE
refactor(FOROME-725): modals compound request

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-721

PR into refactor/FOROME-724

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/components/request-block.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+export const RequestBlock = observer(
+  ({ index, activeRequestIndex, item }: IRequestBlockProps): ReactElement => {
+    const { problemGroups } = modalEditStore
+    return (
+      <div
+        className={cn(
+          'flex justify-between w-full shadow-dark py-2 my-2 px-2 cursor-pointer step-content-area',
+          index === activeRequestIndex ? 'bg-blue-medium' : 'bg-white',
+        )}
+        key={Math.random()}
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/components/request-block.tsx]: why do you need a key here, it's not even an array

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/components/request-block.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        onClick={(e: any) =>
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/components/request-block.tsx]: React.MouseEvent

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      )
+
+      const params = `{"approx":${approx},"state":${
+        stateCondition === '-current-' || !stateCondition
+          ? null
+          : `"${stateCondition}"`
+      },"request":${requestString}}`
+
+      dtreeStore.fetchStatFuncAsync(groupName, params)
+    }
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.tsx:43]: move to store

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+import { getFuncParams } from '@utils/getFuncParams'
+import { getResetType } from '@utils/getResetType'
+import modalEditStore from '../../modal-edit.store'
+import { EditModalButtons } from '../edit-modal-buttons'
+import { RequestBlock } from './components/request-block'
+import { RequestControlButtons } from './components/request-control-buttons'
+import modalCompoundRequestStore from './modal-compound-request.store'
+
+export const ModalCompoundRequest = observer((): ReactElement => {
+  const ref = useRef(null)
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.tsx]: use more descriptive name

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**